WBooks/Views/BookIndex/BookIndexCell.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  BookIndexCellTableViewCell.swift
+//  WBooks
+//
+//  Created by Diego Quirós on 3/12/18.
+//  Copyright © 2018 Wolox. All rights reserved.
+//
+
+import UIKit
+import Core
+
+final internal class BookIndexCell: UITableViewCell, NibLoadable {
+
+    //static let Height: CGFloat = 100.0
+    
+    @IBOutlet weak var bookImage: UIImageView!
+    @IBOutlet weak var bookTitle: UILabel!
+    @IBOutlet weak var bookAuthors: UILabel!
+    @IBOutlet weak var cellContainer: UIView!
+    
+    func configureCell(for viewModel: BookViewModel) {
+        bookTitle.text = viewModel.title
+        bookAuthors.text = viewModel.author
+        configureImage(urlBook: viewModel.imageURL)
+        
+        cellContainer.layer.cornerRadius = 5
+        cellContainer.backgroundColor = UIColor.white
+        cellContainer.layer.shadowOffset = CGSize(width: 0, height: 1)
+        cellContainer.layer.shadowColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.05).cgColor
+        cellContainer.layer.shadowOpacity = 1
+        cellContainer.layer.shadowRadius = 1
+        
+        self.backgroundColor = UIColor(red: 0.92, green: 0.96, blue: 0.98, alpha: 1)
+        
+    }
+}
+
+private extension BookIndexCell {
+    
+    func configureImage(urlBook: URL?) {
+        guard let url = urlBook else { return }
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
```

Comment:
For fetching images you should use an ImageFetcehr that is already provided in Core or Networking. This implementation has 2 drawbacks:

 * First it directly depends on the URLSession, singleton which makes testing this feature more complicated because you won't be able to easily mock the response. We generally don't test controllers directly most of the UI test at a high level is done via UITest which makes real request. My recomendation is that actually fetching the image should be done in the view model which should receive an ImageFetcher via the initializer and have a default parameter with the "real" implementation.

 * Second and the most serious problem is that this is not contemplating the case when a cell is being reused but the image has not being fetched yet. This could happen is there is latency when fetching the image which will happen. In that case you will see that a book suddenly uses an old image. To avoid this you should be able to cancel the fetching tasks.

Here is a solution for problem 2 if you maintain the fetching logic in the view controller.


```swift
final internal class BookIndexCell: UITableViewCell, NibLoadable {

  var imageFetchingTask: URLSessionDataTask? = .none

  override func prepareForReuse() {
    imageFetchingTask?.cancel()
    imageFetchingTask = .none
  }

  func configureImage(urlBook: URL?) {
        guard let url = urlBook else { return }
        imageFetchingTask = URLSession.shared.dataTask(with: url) { (data, response, error) in
            if error != nil {
                print("Failed fetching image \(error.debugDescription)")
                return
            }
            
            guard let response = response as? HTTPURLResponse, response.statusCode == 200 else {
                print("Not a proper HTTPURLResponse or statusCode")
                return
            }
            
            DispatchQueue.main.async {
                self.bookImage.image = UIImage(data: data!)
            }
        }
        imageFetchingTask?.resume()
  }
}
``` 

If you use an ImageFetcher or a soultion based on `SignalProducer`, when you start the producer you need to keep a strong reference to the returned `Disposable` and dispose it on the `prepareForReuse` method to cancel the image fetching and its subscriber. 

You can read about `prepareForReuse` check [this](https://developer.apple.com/documentation/uikit/uitableviewcell/1623223-prepareforreuse).

You can read about memory management in ReactiveSwift [here](https://eliaszsawicki.com/reactiveswift-manage-your-memory/), [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/FrameworkOverview.md#signal-producers) and [here](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/FrameworkOverview.md#disposables)

//cc @aducret